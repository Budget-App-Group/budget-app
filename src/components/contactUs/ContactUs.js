import React, { Component } from "react";
import axios from "axios";
import "./contactUs.scss";

class ContactUs extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      message: "",
      email: "",
    };
  }

  handleInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  nodeMail = () => {
    axios
      .post(`/api/mailer`, {
        name: this.state.name,
        message: this.state.message,
        email: this.state.email,
      })
      .then((res) => {
        alert("Thank you for contacting us, we will reply shortly");
      });
  };

  render() {
    return (
      <div className="contact-main">
        <div className="contact-title">
          <h3>Contact Us</h3>
        </div>
        <form className="contact-form" onSubmit={() => this.nodeMail()}>
          <div className="name-area">
            <label className="message-name" htmlFor="message-name">
              Your Name:
            </label>
            <input
              onChange={this.handleInput}
              name="name"
              className="message-input"
              type="text"
              placeholder="Your Name"
              value={this.state.name}
              required
            />
          </div>

          <div className="email-area">
            <label className="message-email" htmlFor="message-email">
              Your Email:
            </label>
            <input
              onChange={this.handleInput}
              name="email"
              className="message-input"
              type="email"
              placeholder="you@email.com"
              required
              value={this.state.email}
            />
          </div>

          <div className="message-area">
            <label className="message-label" htmlFor="message-input">
              Your Message:
            </label>
            <textarea
              onChange={this.handleInput}
              name="message"
              className="message-box"
              type="text"
              placeholder="Please write your message here"
              value={this.state.message}
              required
            />
          </div>

          <div className="submit-button-container">
            <button type="submit" className="submit-contact-button">
              Send Message
            </button>
          </div>
        </form>
      </div>
    );
  }
}

export default ContactUs;
