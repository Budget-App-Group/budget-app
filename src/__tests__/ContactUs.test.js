import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import ContactUs from "../components/contactUs/ContactUs";
import { MemoryRouter } from "react-router";

it("Join does show Sign In button text when mounted", () => {
  const { container } = render(
    <MemoryRouter>
      <ContactUs />
    </MemoryRouter>
  );

  expect(container.textContent).toContain("Contact Us");
});

//snapshot tests
describe("contact us snapshot tests", () => {
  const component = renderer.create(
    <MemoryRouter>
      <ContactUs />
    </MemoryRouter>
  );
  let tree = component.toJSON();

  test("component Contact us matches snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  test("component Contact us displays a truthy value", () => {
    expect(tree).toBeTruthy();
  });
});
