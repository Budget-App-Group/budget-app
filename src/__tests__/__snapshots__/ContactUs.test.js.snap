// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact us snapshot tests component Contact us matches snapshot 1`] = `
<div
  className="contact-main"
>
  <div
    className="contact-title"
  >
    <h3>
      Contact Us
    </h3>
  </div>
  <form
    className="contact-form"
    onSubmit={[Function]}
  >
    <div
      className="name-area"
    >
      <label
        className="message-name"
        htmlFor="message-name"
      >
        Your Name:
      </label>
      <input
        className="message-input"
        name="name"
        onChange={[Function]}
        placeholder="Your Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="email-area"
    >
      <label
        className="message-email"
        htmlFor="message-email"
      >
        Your Email:
      </label>
      <input
        className="message-input"
        name="email"
        onChange={[Function]}
        placeholder="you@email.com"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="message-area"
    >
      <label
        className="message-label"
        htmlFor="message-input"
      >
        Your Message:
      </label>
      <textarea
        className="message-box"
        name="message"
        onChange={[Function]}
        placeholder="Please write your message here"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="submit-button-container"
    >
      <button
        className="submit-contact-button"
        type="submit"
      >
        Send Message
      </button>
    </div>
  </form>
</div>
`;

exports[`contact us snapshot tests component Join matches snapshot 1`] = `
<div
  className="contact-main"
>
  <div
    className="contact-title"
  >
    <h3>
      Contact Us
    </h3>
  </div>
  <form
    className="contact-form"
    onSubmit={[Function]}
  >
    <div
      className="name-area"
    >
      <label
        className="message-name"
        htmlFor="message-name"
      >
        Your Name:
      </label>
      <input
        className="message-input"
        name="name"
        onChange={[Function]}
        placeholder="Your Name"
        type="text"
        value=""
      />
    </div>
    <div
      className="email-area"
    >
      <label
        className="message-email"
        htmlFor="message-email"
      >
        Your Email:
      </label>
      <input
        className="message-input"
        name="email"
        onChange={[Function]}
        placeholder="you@email.com"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="message-area"
    >
      <label
        className="message-label"
        htmlFor="message-input"
      >
        Your Message:
      </label>
      <textarea
        className="message-box"
        name="message"
        onChange={[Function]}
        placeholder="Please write your message here"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="submit-button-container"
    >
      <button
        className="submit-contact-button"
        type="submit"
      >
        Send Message
      </button>
    </div>
  </form>
</div>
`;
